subsys/llext/llext_mem.c

```diff
@@ -371,13 +376,23 @@ int llext_add_domain(struct llext *ext, struct k_mem_domain *domain)
 #endif
 }
 
+#ifdef CONFIG_HARVARD
+int llext_heap_init(void *instr_mem, size_t instr_bytes,
+		void *data_mem, size_t data_bytes)
+#else
 int llext_heap_init(void *mem, size_t bytes)
```

Comment:
can we keep a single prototype? Just add the missing arguments to `llext_heap_init()` for everybody (LLEXT is still experimental, isn't it?) or if we absolutely want to keep the API, add another function and call it from the current `llext_heap_init()`?

Reply:
I have updated it, but I don't know if I 100% understood what you suggested - let me know if it's right